### PR TITLE
Investigate differences between XuanTie C-series chips and std PLIC

### PR DESCRIPTION
**Pull request by chenzongyao200127**
Investigate differences between XuanTie C-series chips and RISC-V standard PLIC implementation

This investigation compares the XuanTie C-series chips (including C906, C907, C908, C910, C920 R1S6, and C920 R2S3) with the RISC-V standard PLIC implementation. Key findings include:

1. **Priority Support**:
   - XuanTie C-series chips support only the lowest 5 bits of the priority register, allowing 32 priority levels (0-31), consistent with the RISC-V standard.

2. **Interrupt Sources**:
   - Both XuanTie C-series chips and the RISC-V standard support up to 1023 interrupt sources (source 0 is reserved).

3. **Context and Hart Support**:
   - **C906**: Supports 1 core with 2 contexts (Machine and Supervisor modes).
   - **C920 R1S6 & C910**: Support up to 4 cores, each with 2 contexts.
   - **C907, C908 R1S0, & C920 R2S3**: Support up to 256 cores, each with 2 contexts.
   - The RISC-V standard supports up to 15,872 contexts, significantly more than XuanTie implementations.

4. **Access Control**:
   - XuanTie C-series chips introduce a `PLIC_CTRL` register to control access permissions:

5. **Memory Layout**:
   - XuanTie C-series chips follow a similar memory layout to the RISC-V standard but with additional registers for Machine and Supervisor mode configurations.
   - The RISC-V standard uses a more generalized layout with a fixed offset for each context.

Signed-off-by: Zongyao Chen solar1s@163.com

### CODE REVIEW

**Review by luojia65 — APPROVED** [review]

File: 2024/minutes/RISC-V 标准 PLIC 与平头哥玄铁 PLIC.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+| --------------------------------- | --------------- | ---------------- | ------------------------ |
+| 平头哥 C906 PLIC                  | 5 位（0~31）    | 1023             | 1 个核（2 个上下文）     |
+| 玄铁 C920 R1S6 & C910             | 5 位（0~31）    | 1023             | 4 个核（8 个上下文）     |
+| 玄铁 C907 & C908 R1S0 & C920 R2S3 | 5 位（0~31）    | 1023             | 256 个核（512 个上下文） |
+
+## HART 与 Context ID 对应关系
+
+- **玄铁 C906**：每个 hart 有 2 个上下文，Machine Mode 和 Supervisor Mode 分别对应 Context ID 0 和 1。
+- **玄铁 C920 R1S6 & C910**：最多 4 个 hart，每个 hart 有 2 个上下文，Context ID 从 0 开始按顺序分配。
+- **玄铁 C907 & C908 R1S0 & C920 R2S3**：最多 256 个 hart，每个 hart 有 2 个上下文，Context ID 按 `2n` 和 `2n + 1` 的规则分配。
```
Comment (luojia65) [inline 2024/minutes/RISC-V 标准 PLIC 与平头哥玄铁 PLIC.md:250]: For hart id `n`, M mode corresponds to `2n`, S mode corresponds to `2n + 1`

File: 2024/minutes/RISC-V 标准 PLIC 与平头哥玄铁 PLIC.md [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+| Reserved  | -              | -    | -      | -                                            |
+| 0x0206000 | PLIC_H3_MTH    | R/W  | 0x0    | 核 3 机器模式中断阈值寄存器                  |
+| 0x0206004 | PLIC_H3_MCLAIM | R/W  | 0x0    | 核 3 机器模式中断响应/完成寄存器             |
+| Reserved  | -              | -    | -      | -                                            |
+| 0x0207000 | PLIC_H3_STH    | R/W  | 0x0    | 核 3 超级用户模式中断阈值寄存器              |
+| 0x0207004 | PLIC_H3_SCLAIM | R/W  | 0x0    | 核 3 超级用户模式中断响应/完成寄存器         |
+| Reserved  | -              | -    | -      | -                                            |
+
+
+## 玄铁 C907 & C908 R1S0 & C920 R2S3 PLIC
```
Comment (luojia65) [inline 2024/minutes/RISC-V 标准 PLIC 与平头哥玄铁 PLIC.md:176]: There exists PLIC_CTRL register like C906, please re-check.

File: 2024/minutes/RISC-V 标准 PLIC 与平头哥玄铁 PLIC.md (hunk 1 above)
Comment (luojia65) [inline 2024/minutes/RISC-V 标准 PLIC 与平头哥玄铁 PLIC.md:250]: For hart id `n`, M mode corresponds to `2n`, S mode corresponds to `2n + 1`

**Review by luojia65 — CHANGES_REQUESTED** [review]

File: 2024/minutes/RISC-V 标准 PLIC 与平头哥玄铁 PLIC.md (hunk 1 above)
Comment (chenzongyao200127) [reply 2024/minutes/RISC-V 标准 PLIC 与平头哥玄铁 PLIC.md]: Thanks for pointing that out! I've added the related description to clarify the details. Let me know if there's anything else I can help with：)

File: 2024/minutes/RISC-V 标准 PLIC 与平头哥玄铁 PLIC.md (hunk 2 above)
Comment (chenzongyao200127) [reply 2024/minutes/RISC-V 标准 PLIC 与平头哥玄铁 PLIC.md:176]: Thank you for pointing that out! You are correct that the PLIC_CTRL register exists in the C906 and other XuanTie C-series chips. I have rechecked the documentation and confirmed that the PLIC_CTRL register is indeed present across the XuanTie C-series, including C906, C920 R1S6, C910, C907, C908 R1S0, and C920 R2S3.

To ensure clarity, I have added a summary of the interrupt-related registers at the end of the document, which includes the PLIC_CTRL register and its functionality.

**Review by luojia65 — APPROVED** [review]
LGTM! Thanks for contribution.

**Pull request merged**